### DISCUSSION TITLE
Noisy Console warning for .defaultProps in Plugin Ideal Image

### DISCUSSION THREAD

**Discussion by AndrewSepic** [facebook/docusaurus#10775] [General]
Hi @slorber ,  The use of `plugin-ideal-image` on a new version of docusaurus `3.6.3` causes a noisy console warning about the use of defaultProps outside the component function.  

<img width="1434" alt="Screenshot 2024-12-16 at 11 29 41 AM" src="https://github.com/user-attachments/assets/3e481d15-e09c-4c80-aa99-8e9ce172a7a6" />

I don't see this as an 'issue' as it's not actually an error, but I maintain a bunch of customized docusaurus sites and it's nice to have a clear console so debugging is easier when tracking down my own errors.  😉 

I've [PR'd a an update](https://github.com/slorber/docusaurus-react-ideal-image/pull/2) to your `@slorber/react-ideal-image` here if this is something that would be helpful to others as well.

**Comment by slorber** [comment]
Thanks

It's super weird because I'm using IdealImage on our website running 18.3.1 and didn't see this warning anywhere, even on pages using it.

I've fixed it on the Docusaurus site in the React 19 support PR (https://github.com/facebook/docusaurus/pull/10763) but will probably merge your PR instead, since there's a React 18 warning, and we'd be able to fix it by releasing a patch in this package instead of a patch in Docusaurus (simpler for me).

But really curious, can you help me reproduce this warning first to ensure that we properly get rid of it? Are you running `start` or `build`? Are you using the `--dev` CLI option?

**Comment by AndrewSepic** [comment]
Sorry for the slow response here!  I see that you merged my PR. 🙏 Hopefully it does fix it. 🤞  

I was just running `start`

### To reproduce

1. Install a brand new Docusaurus instance
2. Install the `ideal-image-plugin`
3. Run the local dev server `docusaurus start`
5. Navigate to (and/or create) a doc page that has an Image in it using the `<IdealImage/>` component [as shown here](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-ideal-image#usage)

**Reply by slorber** [reply]
I see thanks

It's weird that I haven't seen this on our own website, but the PR fixes it anyway so let's move on
